### PR TITLE
Implementing datasets panel skeleton

### LINKED ISSUE #17: Populate left panel with current datasets

### PR DESCRIPTION
**Pull request by martinRenou**
Fix #17 

Populate the left panel with current datasets + add menu and commands logic for creating new viewers

Screenshot from 2023-04-18 09-56-51

### CODE REVIEW

File: src/controlPanel/plugin.ts [hunk 1]
```diff
@@ -19,18 +19,39 @@ export const controlPanel: JupyterFrontEndPlugin<void> = {
...
-    app.shell.add(controlPanel, 'left', { rank: 2000 });
+
+    shell.add(controlPanel, 'left', { rank: 2000 });
+
+    commands.addCommand('new-viewer', {
+      label: 'New Viewer',
+      iconClass: 'fa fa-chart-bar',
+      mnemonic: 0,
+      execute: () => {
+        console.log('create new viewer for', controlModel.selectedDataset);
```
Comment (martinRenou) [inline src/controlPanel/plugin.ts:47]: This should be implemented in a follow up PR.

File: src/controlPanel/data/datasetsWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        id === this._model.selectedDataset
+          ? 'glue-Control-datasets-item-selected'
+          : ''
+      }`;
+      return (
+        <li id={id} className={className} onClick={this._onClick.bind(this)}>
+          {id}
+        </li>
+      );
+    };
```
Comment (trungleduc) [inline src/controlPanel/data/datasetsWidget.tsx:104]: This function can be defined outside of the `render` method?

**Comment by martinRenou** [comment]
Turning this into draft again. I want to add a bit more to this.

File: src/controlPanel/data/datasetsWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  }
+
+  private _onClick(event: React.MouseEvent) {
+    const target = event.target as HTMLElement;
+    if (this._dataNames.includes(target.id)) {
+      this._model.selectedDataset = target.id;
+    }
+  }
+
+  render() {
```
Comment (brichet) [inline src/controlPanel/data/datasetsWidget.tsx:92]: To avoid lint warning :

```suggestion
  render(): JSX.Element {
```

File: style/base.css [hunk 4]
```diff
@@ -12,6 +12,29 @@
...
+  flex-direction: row;
+  color: var(--jp-ui-font-color1);
+  height: 24px;
+  line-height: 24px;
+  padding: 0 8px;
+  cursor: pointer;
+}
+
+.glue-Control-datasets-item:hover {
+  background-color: var(--jp-layout-color2);
```
Comment (brichet) [inline style/base.css:31]: Maybe the color could be different when selected and hover.

**Review by brichet — APPROVED** [review]
Thanks @martinRenou, this look good to me.
I just have 2 comments about lint and style.

We should add an issue to implement the drag and drop feature associated to the `new-viewer` command.

File: style/base.css (hunk 4 above)
Comment (martinRenou) [reply style/base.css:31]: Thanks! I looked into what the filebrowser widget does for selections and applied the same styling.

Screenshot from 2023-04-18 11-01-44

**Comment by martinRenou** [comment]
Thanks both! I took your comments into account

**Comment by martinRenou** [comment]
I also changed the implementation to use a sub-menu:

Screenshot from 2023-04-18 11-03-11

**Review by trungleduc — APPROVED** [review]

**Comment by trungleduc** [comment]
Thanks @martinRenou

**Pull request merged**